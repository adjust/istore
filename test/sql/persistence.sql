BEGIN;
CREATE TABLE time_test AS SELECT '2014-01-01 12:00:00'::aj_time, 1 as num;
SELECT * FROM time_test;
UPDATE time_test SET num = 2;
SELECT * FROM time_test;
ROLLBACK;

BEGIN;
CREATE TABLE date_test AS SELECT '2014-01-01'::aj_date, 1 as num;
SELECT * FROM date_test;
UPDATE date_test SET num = 2;
SELECT * FROM date_test;
ROLLBACK;

BEGIN;
CREATE TABLE country_test AS SELECT 'de'::country, 1 as num;
SELECT * FROM country_test;
UPDATE country_test SET num = 2;
SELECT * FROM country_test;
ROLLBACK;

BEGIN;
CREATE TABLE os_name_test AS SELECT 'ios'::os_name, 1 as num;
SELECT * FROM os_name_test;
UPDATE os_name_test SET num = 2;
SELECT * FROM os_name_test;
ROLLBACK;

BEGIN;
CREATE TABLE device_type_test AS SELECT 'phone'::device_type, 1 as num;
SELECT * FROM device_type_test;
UPDATE device_type_test SET num = 2;
SELECT * FROM device_type_test;
ROLLBACK;