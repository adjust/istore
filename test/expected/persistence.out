BEGIN;
-- persistence should persist countries;
-- ./spec/persistence_spec.rb:8;
CREATE EXTENSION istore;
ERROR:  type "istore" already exists
CREATE TABLE country_test AS SELECT 'de'::country, 1 as num;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT * FROM country_test;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
UPDATE country_test SET num = 2;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT * FROM country_test;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK;
BEGIN;
-- persistence should persist os_names;
-- ./spec/persistence_spec.rb:15;
CREATE EXTENSION istore;
ERROR:  type "istore" already exists
CREATE TABLE os_name_test AS SELECT 'ios'::os_name, 1 as num;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT * FROM os_name_test;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
UPDATE os_name_test SET num = 2;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT * FROM os_name_test;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK;
BEGIN;
-- persistence should persist device_type;
-- ./spec/persistence_spec.rb:22;
CREATE EXTENSION istore;
ERROR:  type "istore" already exists
CREATE TABLE device_type_test AS SELECT 'phone'::device_type, 1 as num;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT * FROM device_type_test;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
UPDATE device_type_test SET num = 2;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT * FROM device_type_test;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK;
